includes/abstracts/llms-abstract-controller-user-engagements.php

```diff
@@ -0,0 +1,234 @@
+		// Invalid actions return a WP_Error for testing purposes and are not displayed to the user.
+		$actions = array(
+			'sync_one'  => "sync_awarded_$this->engagement_type",
+			'sync_many' => "sync_awarded_{$this->engagement_type}s",
+		);
+		if ( ! isset( $_GET['action'] ) ) {
+			return new WP_Error(
+				"llms-sync-awarded-{$this->engagement_type}s-missing-action",
+				__( 'Sorry, you have not provided any actions.', 'lifterlms' )
+			);
+		} elseif ( ! in_array( $_GET['action'], $actions ) ) {
+			return new WP_Error(
+				"llms-sync-awarded-{$this->engagement_type}s-invalid-action",
+				__( 'You\'re trying to perform an invalid action.', 'lifterlms' )
+			);
+		}
+
+		// Verify nonce.
+		if ( ! llms_verify_nonce(
+			"_llms_{$this->engagement_type}_sync_actions_nonce",
+			"llms-$this->engagement_type-sync-actions",
+			'GET'
+		) ) {
+			$result = new WP_Error(
+				"llms-sync-awarded-{$this->engagement_type}s-invalid-nonce",
+				sprintf(
+					/* translators: 1: plural lowercase awarded engagement type, 2: plural uppercase first letter awarded engagement type */
+					__( 'Sorry, you are not allowed to sync %1$s.', 'lifterlms' ),
+					$this->get_engagement_type_name( self::PLURAL, self::CASE_LOWER, self::AWARDED ),
+					$this->get_engagement_type_name( self::PLURAL, self::CASE_UPPER_FIRST, self::AWARDED )
+				)
+			);
+			( new LLMS_Meta_Box_Award_Engagement_Submit() )->add_error( $result );
+
+			return $result;
+		}
+
+		$engagement_id = llms_filter_input( INPUT_GET, 'post', FILTER_SANITIZE_NUMBER_INT );
+
+		if ( $_GET['action'] === $actions['sync_one'] ) {
+
+			if ( empty( $engagement_id ) ) {
+				$result = new WP_Error(
+					"llms-sync-missing-awarded-$this->engagement_type-id",
+					sprintf(
+						/* translators: 1: singular lowercase awarded engagement type, 2: singular uppercase first letter awarded engagement type */
+						_x( 'Sorry, you need to provide a valid %1$s ID.', 'awarded', 'lifterlms' ),
+						$this->get_engagement_type_name( self::SINGULAR, self::CASE_LOWER, self::AWARDED ),
+						$this->get_engagement_type_name( self::SINGULAR, self::CASE_UPPER_FIRST, self::AWARDED )
+					)
+				);
+			} else {
+				$result = $this->sync_awarded_engagement( $engagement_id );
+			}
+		} elseif ( $_GET['action'] === $actions['sync_many'] ) {
+
+			if ( empty( $engagement_id ) ) {
+				$result = new WP_Error(
+					"llms-sync-missing-{$this->engagement_type}-template-id",
+					sprintf(
+						/* translators: 1: singular lowercase engagement template type, 2: singular uppercase first letter engagement template type */
+						_x( 'Sorry, you need to provide a valid %1$s ID.', 'template', 'lifterlms' ),
+						$this->get_engagement_type_name( self::SINGULAR, self::CASE_LOWER, self::TEMPLATE ),
+						$this->get_engagement_type_name( self::SINGULAR, self::CASE_UPPER_FIRST, self::TEMPLATE )
+					)
+				);
+			} else {
+				$result = $this->sync_awarded_engagements( $engagement_id );
+			}
+		} else {
+			$result = null;
```

Comment:
```suggestion
```

We can't get here because $action must be in the $actions array in order to proceed past the initial validation

Reply:
True, but some code sniffing tools (I'm looking at you PhpStorm!) can't figure that out and report that `$result` may be undefined. Another option is to set `$result = null` before that `if` block.